h2/src/main/org/h2/command/Parser.java

```diff
@@ -9926,6 +9926,17 @@ private Prepared parseAlterTableCompatibility(Schema schema, String tableName, b
                 command.setOldColumnName(columnName);
                 command.setNewColumnName(newColumnName);
                 return command;
+            } else if (readIf("CONVERT")) {
```

Comment:
this needs a 

    && database.getMode().getEnum() == ModeEnum.MySQL

so it doesn't accidentally start "working" for other database testing

Reply:
Sorry, this extra condition isn't needed and actually should'n be used.

We have `if (mode.alterTableExtensionsMySQL)` above, so code is already executed only in MySQL and MariaDB compatibility modes. It looks like MariaDB also supports this command, so there are no reasons to restrict this syntax only to MySQL compatibility mode.